aip/auth/4110.md

```diff
@@ -127,10 +127,21 @@ digraph d_front_back {
     1. If the credential is [an external account][8] JSON, go to step (4)
     1. If the credential is unknown type, return an error saying that _[END]_
   1. Credentials not found _[END]_
-1. **Check Google Virtual Machine Credentials**
-  1. If true, use the [virtual machine flow][3] to fetch an auth token associated with the current environment
-    1. If target audience is provided by the developer, get an [identity token][7] _[END]_
-    1. Otherwise, get an access token _[END]_
+1. **Check workload credentials (on GCE, GKE, GAE and Serverless)**
+  1. If true,
+    1. If identity binding is enabled, by meeting the requirements in
+       [mTLS Token Binding][9], use the mTLS Token Binding flow to fetch an
+       identity-bound access token _[END]_
+    1. If only one of the **"workload_identity_provider"** and
```

Comment:
But AIP 4119 says it's okay for the `service_account_email` to be absent.

Reply:
This entire item is not relevant anymore. If "workload_identity_provider" is not set, then we would not get a bound token. If it is set but "authenticate_as_identity_type", we default to "gsa". If "service_account_email" is not set, we get its value from MDS. In all cases, we have a fallback and we don't need to return an error.

Reply:
Good point.